File: frontend/src/types/fund.ts
export type TransactionType = 'deposit' | 'withdrawal'

export interface FundTransaction {
  id: string
  amount: number
  transaction_type: TransactionType
  description: string
  related_settlement_id?: string
  transaction_date: string
  created_at: string
}

export interface CreateDepositRequest {
  amount: number
  description: string
  date?: string
}

export interface CreateWithdrawalRequest {
  amount: number
  description: string
  date?: string
}

export interface FundBalance {
  balance: number
}

export interface FundStats {
  total_deposits: number
  total_withdrawals: number
  settlement_deposits: number
  balance: number
}

File: backend/internal/model/fund.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type FundTransaction struct {
	ID                  uuid.UUID  `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	Amount              int        `gorm:"not null" json:"amount"` // Amount in VND
	TransactionType     string     `gorm:"type:varchar(20);not null" json:"transaction_type"` // "deposit" or "withdrawal"
	Description         string     `gorm:"type:text" json:"description"`
	RelatedSettlementID *uuid.UUID `gorm:"type:uuid" json:"related_settlement_id,omitempty"`
	TransactionDate     time.Time  `gorm:"default:now()" json:"transaction_date"`
	CreatedAt           time.Time  `json:"created_at"`
}

func (FundTransaction) TableName() string {
	return "fund_transactions"
}
